powerline/segments/i3wm.py

```diff
@@ -54,12 +54,44 @@ def workspaces(pl, segment_info, only_show=None, output=None, strip=0):
 		'contents': w['name'][min(len(w['name']), strip):],
 		'highlight_groups': calcgrp(w)
 	} for w in conn.get_workspaces()
-		if (not only_show or any(w[typ] for typ in only_show))
-		and (not output or w['output'] == output)
+		if ((not only_show or any(w[typ] for typ in only_show))
+			and (not output or w['output'] == output))
 	]
 
 
 @requires_segment_info
+def workspace(pl, segment_info, workspace=None, strip=0):
```

Comment:
@ZyX-I but this will also split workspaces if they contain a colon somewhere else (like `a:bc`), i3bar only splits if its a number followed by a colon at the beginning of the string.